### PR TITLE
fix: modify the register to sentinel

### PR DESCRIPTION
**Pull request by DWJ-Squirtle**

### CODE REVIEW

File: addons/redis/scripts/redis-register-to-sentinel.sh [hunk 1]
```diff
@@ -164,9 +180,20 @@ register_to_sentinel() {
   call_func_with_retry 3 5 check_connectivity "$sentinel_host" "$sentinel_port" "$SENTINEL_PASSWORD" || exit 1
   call_func_with_retry 3 5 check_connectivity "$redis_primary_host" "$redis_primary_port" "$REDIS_DEFAULT_PASSWORD" || exit 1
 
-  # Register and configure the Redis primary to redis sentinel
-  sentinel_commands=("monitor" "down-after-milliseconds" "failover-timeout" "parallel-syncs" "auth-user" "auth-pass")
-  for cmd in "${sentinel_commands[@]}"
+# Check if Sentinel is already monitoring the Redis primary
+  master_addr=$(call_func_with_retry 3 5 get_master_addr_by_name "$sentinel_host" "$sentinel_port" "SENTINEL get-master-addr-by-name $master_name")
+
+  if is_empty "$master_addr"; then
```
Comment (Y-Rookie) [inline addons/redis/scripts/redis-register-to-sentinel.sh:186]: $master_addr will also not be empty when get_master_addr_by_name execute failed

File: addons/redis/scripts/redis-register-to-sentinel.sh [hunk 2]
```diff
@@ -164,9 +184,20 @@ register_to_sentinel() {
   call_func_with_retry 3 5 check_connectivity "$sentinel_host" "$sentinel_port" "$SENTINEL_PASSWORD" || exit 1
   call_func_with_retry 3 5 check_connectivity "$redis_primary_host" "$redis_primary_port" "$REDIS_DEFAULT_PASSWORD" || exit 1
 
-  # Register and configure the Redis primary to redis sentinel
-  sentinel_commands=("monitor" "down-after-milliseconds" "failover-timeout" "parallel-syncs" "auth-user" "auth-pass")
-  for cmd in "${sentinel_commands[@]}"
+# Check if Sentinel is already monitoring the Redis primary
```
Comment (zjx20) [inline addons/redis/scripts/redis-register-to-sentinel.sh:187]: indention

File: addons/redis/scripts/redis-register-to-sentinel.sh [hunk 3]
```diff
@@ -164,9 +184,20 @@ register_to_sentinel() {
...
+  master_addr=$(get_master_addr_by_name "$sentinel_host" "$sentinel_port" "SENTINEL get-master-addr-by-name $master_name")
```
Comment (zjx20) [inline addons/redis/scripts/redis-register-to-sentinel.sh:188]: `exit 1` if get_master_addr_by_name returns 1?

File: addons/redis/scripts/redis-register-to-sentinel.sh [hunk 4]
```diff
@@ -164,9 +184,20 @@ register_to_sentinel() {
...
+    call_func_with_retry 3 5 execute_sentinel_sub_command "$sentinel_host" "$sentinel_port" "$sentinel_monitor_cmd"
+  else
+    echo "Sentinel is already monitoring $master_name at $master_addr. Skipping monitor registration."
+  fi
+  #configure the Redis primary with Sentinel
+  sentinel_configure_commands=("down-after-milliseconds" "failover-timeout" "parallel-syncs" "auth-user" "auth-pass")
+  for cmd in "${sentinel_configure_commands[@]}"
   do
     sentinel_cli_cmd=$(construct_sentinel_sub_command "$cmd" "$master_name" "$redis_primary_host" "$redis_primary_port")
     call_func_with_retry 3 5 execute_sentinel_sub_command "$sentinel_host" "$sentinel_port" "$sentinel_cli_cmd"
```
Comment (zjx20) [inline addons/redis/scripts/redis-register-to-sentinel.sh:203]: `call_func_with_retry xxx || exit 1` ?

What do you think? @Y-Rookie

File: addons/redis/scripts/redis-register-to-sentinel.sh [hunk 5]
```diff
@@ -150,6 +150,26 @@ execute_sentinel_sub_command() {
...
+  local sentinel_port=$2
+  local command=$3
+  local output
+  output=$(redis-cli -h "$sentinel_host" -p "$sentinel_port" -a "$SENTINEL_PASSWORD" $command)
+  local status=$?
+  if [ $status -ne 0 ]; then
+    echo "Command failed with status $status." >&2
+    return 1
+  fi
+  if echo "$output" | grep -E '^[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+\.default\.svc$' > /dev/null || is_empty "$output"; then
```
Comment (Y-Rookie) [inline addons/redis/scripts/redis-register-to-sentinel.sh:164]: how about NodePort mode? the output will not match regexp patten ?

File: addons/redis/scripts/redis-register-to-sentinel.sh (hunk 4 above)
Comment (Y-Rookie) [reply addons/redis/scripts/redis-register-to-sentinel.sh:205]: > `call_func_with_retry xxx || exit 1` ?
> 
> What do you think? @Y-Rookie

it  would be better @DWJ-Squirtle

**Review by Y-Rookie — APPROVED** [review]

**Review by zjx20 — APPROVED** [review]

**Pull request merged**